From 2731d5b34ab1db7d97691eeade8b1d7535ea5a34 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Fri, 9 Mar 2018 17:28:07 +0000
Subject: [PATCH 4/4] lan78xx: Change LEDs to include 10Mb activity

The default LED modes put 1000Mb/activity on orange and 100Mb/activity
on green, but this leaves no indication for a 10Mb link. Change the
defaults to put 10Mb/100Mb/activity on green.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 drivers/net/usb/lan78xx.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index 25f61d7..885dab5 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -1849,6 +1849,7 @@ static int lan78xx_phy_init(struct lan78xx_net *dev)
 {
 	int ret;
 	u32 mii_adv;
+	u32 led_modes;
 	struct phy_device *phydev = dev->net->phydev;
 
 	phydev = phy_find_first(dev->mdiobus);
@@ -1888,6 +1889,19 @@ static int lan78xx_phy_init(struct lan78xx_net *dev)
 	mii_adv = (u32)mii_advertise_flowctrl(dev->fc_request_control);
 	phydev->advertising |= mii_adv_to_ethtool_adv_t(mii_adv);
 
+	/* Change LED defaults:
+	 *   orange = link1000/activity
+	 *   green  = link10/link100/activity
+	 * led: 0=link/activity          1=link1000/activity
+	 *      2=link100/activity       3=link10/activity
+	 *      4=link100/1000/activity  5=link10/1000/activity
+	 *      6=link10/100/activity    14=off    15=on
+	 */
+	led_modes = phy_read(phydev, 0x1d);
+	led_modes &= ~0xff;
+	led_modes |= (1 << 0) | (6 << 4);
+	(void)phy_write(phydev, 0x1d, led_modes);
+	
 	genphy_config_aneg(phydev);
 
 	dev->fc_autoneg = phydev->autoneg;
-- 
2.7.4

